### PR TITLE
Adds initial llama3 model

### PR DESCRIPTION
**Pull request by pstjohn**
Adds an initial llama3 model using TransformerLayers.

This will be tricky to run in CI, since we don't want to download the full 8B model weights every run.

### CODE REVIEW

**Review by jstjohn — APPROVED** [review]

**Comment by pstjohn** [comment]
force-merging with John St. John's approval and CI passing.

File: bionemo-recipes/models/llama3/modeling_llama_te.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    def __init__(self, config: LlamaConfig):
+        """Initialize the NVLlama model."""
+        super().__init__(config)
+        self.padding_idx = config.pad_token_id
+        self.vocab_size = config.vocab_size
+
+        self.embed_tokens = nn.Embedding(config.vocab_size, config.hidden_size, self.padding_idx, dtype=config.dtype)
+        self.layers = nn.ModuleList(
+            [
+                transformer_engine.pytorch.TransformerLayer(
```
Comment (jomitchellnv) [inline bionemo-recipes/models/llama3/modeling_llama_te.py:53]: no attn mask type?
